server/lib/prune-newick.js

```diff
@@ -0,0 +1,163 @@
+/*
+This file will prune a newick tree to remove genes that are too dissimilar. 
+*/
+
+const hammingDistance = require('../hamdis/hamming-distance')
+const {parseFasta} = require('../formats/fasta/parse')
+
+module.exports.pruneOutliers = pruneOutliers
+
+function pruneOutliers(newick, alignedFasta){
+
+	const fastaData = parseFasta(alignedFasta)
+	// Build a dict map of species -> sequence 
+	let sequenceMap = {}
+	for(let obj of fastaData){
+		sequenceMap[obj.species] = obj.sequence
+	}
+
+	let leafNodes = []
+	function GetLeaves(node) {
+		if (node.branchset) {
+			node.branchset.forEach(GetLeaves)
+		} else {
+			leafNodes.push(node)
+		}
+	}
+	// Compute average and standard deviation of all the pairs of distances 
+	GetLeaves(newick)
+	let average = 0
+	let totalDistances = []
+	let distCache = {}
```

Comment:
`distCache` might be a bit misleading; just `distances`?  Also… it's an `Object` but you're [using it as an `Array`](https://github.com/hybsearch/hybsearch/pull/63/files#diff-4da64203a2b963e143cd55873da19374R36), which is perfectly valid but begs the question of why it's not an `Array`.